### DISCUSSION TITLE
Bottle example in cascade studio.

### DISCUSSION THREAD

**Discussion by mmiscool** [zalo/CascadeStudio#112] [Q&A]
Hello,

I wanted to see if there was a working bottle example made in cascade studio. I tried copying from the opencascade.js-examples but it seems like library is different.

**Comment by raydeleu** [comment]
I did not find a version of the OpenCascade tutorial bottle in CascadeStudio. It would be possible to model this, but be aware that some functions available in OpenCascade are not included in CascadeStudio. So you would have to work around these:

* the base shape requires a mirror function according to the tutorial. I think that the mirror function was proposed just this week. You can achieve the same thing by adding points to your sketch at the right location. 

* to create a hollow shape OpenCascade has a function where you select a face of a closed surface to be removed and add thickness to the shape inward or outward. You can achieve something similar by using an offset to the shape and subtracting this from the original shape (see my manual that I announced in this forum). 

* the thread is difficult to achieve. In my manual I refer to a solution proposed earlier on cadhub.xyz. see https://github.com/zalo/CascadeStudio/discussions/48

Of course you can also try to create your own function that exposes the OpenCascade functionality. I tried this for thin-walled objects but did not succeed, so it is not always simple to do.

**Comment by raydeleu** [comment]
See 

https://github.com/raydeleu/CascadeStudioManual/blob/main/readme.adoc

for examples how to add your own functionality to CascadeStudio.

**Comment by gulamf** [comment]
here is an example of the bottle that may be helpful.
https://github.com/zalo/CascadeStudio/issues/33

Scroll down to find the working example.
